### ISSUE TITLE
Android: Make switch of mode easier

### ISSUE THREAD

**Issue by rugk** [mapillary/mapillary_issues#2690]
Switching between the recording modes (front, right left, back; compass) should be faster, as you may want to change the recording direction quite often when *not* recording while driving with a car. So by foot, e.g.

E.g. I think showing the camera image in that screen there takes some time. And it is useless. I just don't need a "camera preview" there, I just need to change the option.

**Comment by katrinhumal** [comment]
@rugk Thanks for your feedback! I'm interested to learn more about when you feel like you need to change the compass angle quite often. The standard capturing case would be that you actually keep your phone facing the same direction with respect to your movement (i.e. hold it in front of you, or 90 degrees to the side, while you move forward - but not waving your camera from facing left to right, back to front etc.).

**Comment by rugk** [comment]
Yeah that is the default when you drive or so, but when you walk, you may actually change your direction very fast when you see something interesting (a junction, a street sign, …) on the right or left, which was not captured by purely taking images of the front.

**Comment by katrinhumal** [comment]
@rugk Thanks for explaining! Theoretically, choosing the compass should be the appropriate option for this case. Provided that it works (but when walking, there should not be much interference - in a car, for example, the metal can easily confuse the compass).

**Comment by rugk** [comment]
My compass is always confused😂

**Comment by katrinhumal** [comment]
Darn 😁 Depending on how often this actually occurs in practice that you feel the need to change direction (but your compass doesn't capture it), you could also consider editing this post-upload?

**Comment by rugk** [comment]
Yes, I know.

**Comment by kaitu** [comment]
@rugk: have you tried to calibrate the compass before you start capturing? It's suggested that rotating the phone around all of its axes with compass on should help fix the compass reading, I think that there are even apps whici assist in the process. Also, I find that the compass reacts pretty slowly to changes of directions. It might help to wait a while after you have changed direction, to give time to the compass reading to settle. If you're in automatic shooting mode, just let it take more pictures aiming at the same target, and keep the last one.

@katrinhumal: post-upload editing is only viable when you know the correct direction from other sources. Otherwise, post capture setting of direction to direction of travel is always doable when a gps track (possibly just a sequence of georeferenced images) is available, there is no need to do it within the application. When walking and intentionally adjusting the direction to opportunity targets, I advice to always set the direction to compass reading.

**Comment by rugk** [comment]
Yeah, I think the compass works better when I hold the phone horizontally, but in this case I cannot capture pictures.

**Comment by katrinhumal** [comment]
_mapillary_issues is no longer an active repository.
Let’s continue discussing features in our forum! We will import this issue as topic to the Mapillary Forum in the coming weeks. We’ll notify here once it’s been done. See you there!_
